### DISCUSSION TITLE
RPI Pico Hat for ease of life and improved measurements?

### DISCUSSION THREAD

**Discussion by Blaster1920** [fhdm-dev/scoppy#38] [Scoppy Front end designs]
Hi all,
just came across this wonderful project.

I was wondering if a simple hat could be designed so that we'd have +-5V acquisition and a couple of BNC males to directly connect the probes to the channels on the pico.
I never had the chance to make such a circuit before, but if others are interested/ more knowledgeable, I can try to put something down and have a cheap PCB made.

Let me know

**Comment by fhdm-dev** [comment]
> Hi all, just came across this wonderful project.
> 
> I was wondering if a simple hat could be designed so that we'd have +-5V acquisition and a couple of BNC males to directly connect the probes to the channels on the pico. I never had the chance to make such a circuit before, but if others are interested/ more knowledgeable, I can try to put something down and have a cheap PCB made.
> 
> Let me know

Great idea. I've made a few crude attempts at designing some [front end circuitry](https://oscilloscope.fhdm.xyz/wiki/Analog-Front-End-Examples) but not a PCB or Hat, so I'd be interested to see what you come up with.

**Reply by Blaster1920** [reply]
> 
> Great idea. I've made a few crude attempts at designing some [front end circuitry](https://oscilloscope.fhdm.xyz/wiki/Analog-Front-End-Examples) but not a PCB or Hat, so I'd be interested to see what you come up with.

Alright, I'll try to reproduce circuit 4 then.
Do you think the other gpios could be used for digital signal decoding? We can harness them in a universal connector for logic probes.
About the hazard of connecting a good mobile, what if we simply add a BT module like the HM-19? The circuit can still be powered using a normal USB powerbank.

Let me know what you think.

**Reply by fhdm-dev** [reply]
> Alright, I'll try to reproduce circuit 4 then.

Cool. Since publishing that circuit I found that using (substantially) lower values for the input resistors can reduce noise (especially at higher sensitivies) - at the expense of lower input impedence of course.

> Do you think the other gpios could be used for digital signal decoding? We can harness them in a universal connector for logic probes. 

Definitely. GPIO6 to GPIO13 are the input GPIOs used when the app is in [_Logic Analyzer_](https://oscilloscope.fhdm.xyz/app-help/Oscilloscope-and-Logic-Analyzer-Modes) mode.

> About the hazard of connecting a good mobile, what if we simply add a BT module like the HM-19? The circuit can still be powered using a normal USB powerbank.

BT should be possible though I'm not sure how difficult this would be on the Android side of things - I'll look into it. Wi-Fi is also another option - eg. via an ESP32.

**Reply by Blaster1920** [reply]
What do you think about the [ADA4850-2](https://www.analog.com/en/products/ada4850-2.html), it's very very fast, I think we would have unity gain for all the mainly used bandwidth, and it's rail-to-rail.

**Reply by fhdm-dev** [reply]
> What do you think about the [ADA4850-2](https://www.analog.com/en/products/ada4850-2.html), it's very very fast, I think we would have unity gain for all the mainly used bandwidth, and it's rail-to-rail.

Looks very impressive for the price. I'm no expert but I don't think it needs to be this fast and so you might be able to find something even cheaper. At a sample rate of 500ksps, Scoppy is only really useful up to a frequency of about 50kHz. I'm not sure what gain you're planning on but even a gain of 20 means a GBP of 10Mhz should be enough .... I think.

Having rail-to-rail outputs is a definite plus. Pity the inputs aren't RtoR. Using a 3.3V supply the input range is limited to 0-1.1V. 

One other downside of this op amp is that it only comes in an LFSCP package. That would make it difficult to prototype on a breadboard or perf board for someone like me.

I'm considering the MCP6292 as an alternative to the LM358 I've been using so far. It's nowhere near as capable as the ADS4850-2 but is cheaper.

**Comment by MaxVandenbussche** [comment]
Hello, I just learned about this project from the article on Hackaday.
I tried out some stuff with a pico and an android device and I was delighted.
Soon we'll starting the development of a small PCB with a nice front-end.

**Reply by fhdm-dev** [reply]
Awesome :smile:
Looking forward to seeing your PCB.

**Comment by helvo52** [comment]
My first Scoppy HAT for Rp2040 Pico
... it was a bad choice to assemble with wired parts, it's getting a little too tight....

I described a lot of functions in the circuit diagram.
[pico-scoppy-2ch-ACDC-4fix2.pdf](https://github.com/fhdm-dev/scoppy/files/8344783/pico-scoppy-2ch-ACDC-4fix2.pdf)


It has 2CH Input with LM358 supplied with +5V/-2,5V , 1CH with Voltage range switch CD4052.
its not complete tested, only calculated with LtSPICE.
![20220316 pico-hat scoppy1l](https://user-images.githubusercontent.com/34276373/159995221-d77e3e4b-ab58-4e4c-bf33-66b90344b66e.jpg)
![20220316 pico-hat scoppy2l](https://user-images.githubusercontent.com/34276373/159995527-4afc4ce5-6ace-44a0-bdbb-685999304c1f.jpg)

First i send only Fotos and Schematic and Eagle 6.6 File
[pico-scoppy-2ch-ACDC-4fix2.zip](https://github.com/fhdm-dev/scoppy/files/8344786/pico-scoppy-2ch-ACDC-4fix2.zip)
s

**Reply by fhdm-dev** [reply]
That's awesome! I'm impressed by the way you are able to fit so many components in such a small area.
Thanks for sharing :smile:
